chat/src/main/java/com/github/twitch4j/chat/TwitchChat.java

```diff
@@ -327,6 +339,17 @@ public TwitchChat(EventManager eventManager, CredentialManager credentialManager
 
         // register event handler
         eventManager.onEvent(ChannelMessageEvent.class, this::onChannelMessage);
+        eventManager.onEvent(IRCMessageEvent.class, event -> {
+            // we get at least one room state event with channel name + id when we join a channel, so we cache that to provide channel id + name for all events
+            if (event.getCommandType().equals("ROOMSTATE")) {
+                if (event.getChannelId() != null && !channelIdToChannelName.containsKey(event.getChannelId())) {
+                    channelIdToChannelName.put(event.getChannelId(), event.getChannelName().get());
+                }
+                if (event.getChannelName().isPresent() && !channelNameToChannelId.containsKey(event.getChannelName().get())) {
+                    channelNameToChannelId.put(event.getChannelName().get(), event.getChannelId());
```

Comment:
If someone sends a `JOIN` and a `PART` shortly after but the `ROOMSTATE` arrives after the `PART` has already been issued, `currentChannels` will no longer be tracking this channel but we are still caching these mappings (that will stick around in memory until another PART is issued)

Reply:
I didn't think about this case at all, i added a condition in there to only track it if the channel is still in the currentChannels list (ie. we didn't leave yet)

Reply:
Think there's a small logic error with the currentChannels check (we're passing a boolean rather than a String 😛), here's a possible replacement:

```java
if (event.getChannelId() != null) {
    event.getChannelName().map(String::toLowerCase).filter(currentChannels::contains).ifPresent(name -> {
        channelNameToChannelId.put(name, event.getChannelId());
        channelIdToChannelName.put(event.getChannelId(), name);
    });
}
```

Also, we may need some synchronization here in case a `PART` on the same channel occurs concurrently to the `ROOMSTATE` like so:

1. `currentChannels::contains` here is true
2. `currentChannels.remove(lowerChannelName)` in `leaveChannel` is true
3. `leaveChannel` somehow continues execution before `ifPresent` is called; as a result, the mappings have been removed from the cache
4. The `ifPresent` block is finally called and the mappings are added back to the cache

Lmk if this seems too fantastical